src/AspNetCoreModuleV2/CommonLib/FileOutputManager.cpp

```diff
@@ -0,0 +1,188 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+#include "stdafx.h"
+
+
+FileOutputManager::FileOutputManager()
+    : m_pStdOutFile(NULL)
+{
+}
+
+FileOutputManager::~FileOutputManager()
+{
+    HANDLE   handle = NULL;
+    WIN32_FIND_DATA fileData;
+
+    if (m_pStdOutFile != NULL)
+    {
+        m_Timer.CancelTimer();
+        _close(_fileno(m_pStdOutFile));
+        m_pStdOutFile = NULL;
+    }
+
+    // delete empty log file
+    handle = FindFirstFile(m_struLogFilePath.QueryStr(), &fileData);
+    if (handle != INVALID_HANDLE_VALUE &&
+        fileData.nFileSizeHigh == 0 &&
+        fileData.nFileSizeLow == 0) // skip check of nFileSizeHigh
+    {
+        FindClose(handle);
+        // no need to check whether the deletion succeeds
+        // as nothing can be done
+        DeleteFile(m_struLogFilePath.QueryStr());
+    }
+
+    if (_dup2(m_fdStdOut, _fileno(stdout)) == -1)
+    {
+        return;
+    }
+
+    if (_dup2(m_fdStdErr, _fileno(stderr)) == -1)
+    {
+        return;
+    }
+}
+
+HRESULT
+FileOutputManager::Initialize(PCWSTR pwzStdOutLogFileName, PCWSTR pwzApplicationPath)
+{
+    HRESULT hr = S_OK;
+
+    if (FAILED(hr = m_wsApplicationPath.Copy(pwzApplicationPath)))
```

Comment:
the code can be `    if (SUCCEED(hr = m_wsApplicationPath.Copy(pwzApplicationPath)))
    {
        hr = m_wsStdOutLogFileName.Copy(pwzStdOutLogFileName));
    }
    return hr`
it saves a check